src/sentry/utils/sdk_crashes/event_stripper.py

```diff
@@ -1,26 +1,64 @@
-from typing import Any, Mapping, Sequence, Tuple
+from enum import Enum, auto
+from typing import Any, Dict, Mapping, Optional, Sequence
 
 from sentry.eventstore.models import Event
 from sentry.utils.safe import get_path
 from sentry.utils.sdk_crashes.sdk_crash_detector import SDKCrashDetector
 
-ALLOWED_EVENT_KEYS = {
-    "type",
-    "datetime",
-    "timestamp",
-    "platform",
-    "sdk",
-    "level",
-    "logger",
-    "exception",
-    "debug_meta",
-    "contexts",
+
+class Allow(Enum):
+    def __init__(self, explanation: str = "") -> None:
+        self.explanation = explanation
+
+    """Keeps the event data if it is of type str, int, float, bool."""
+    SIMPLE_TYPE = auto()
+
+    """Keeps the event data no matter the type."""
+    ALL = auto()
+
+    """
+    Doesn't keep the event data no matter the type. This can be used to explicitly
+    specify that data should be removed with an explanation.
+    """
+    NEVER = auto()
+
+    def with_explanation(self, explanation: str) -> "Allow":
+        self.explanation = explanation
+        return self
+
+
+EVENT_DATA_ALLOWLIST = {
+    "type": Allow.SIMPLE_TYPE,
+    "datetime": Allow.SIMPLE_TYPE,
+    "timestamp": Allow.SIMPLE_TYPE,
+    "platform": Allow.SIMPLE_TYPE,
+    "sdk": {
+        "name": Allow.SIMPLE_TYPE,
+        "version": Allow.SIMPLE_TYPE,
+        "integrations": Allow.NEVER.with_explanation("Users can add their own integrations."),
+    },
+    "exception": Allow.ALL.with_explanation("We strip the exception data separately."),
```

Comment:
This still seems dangerous. There could be fields in `exception` that we don't want to keep, right?

Reply:
Yes, indeed. I created an issue (https://github.com/getsentry/sentry/issues/50710) to solve this in a subsequent PR.